signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java

```diff
@@ -38,13 +42,15 @@
             MetricMetadata metricMetadata,
             String defaultSourceName,
             String sourceDimension,
-            boolean injectCurrentTimestamp) {
+            boolean injectCurrentTimestamp,
+            ImmutableMap<String,String> defaultDimensions) {
```

Comment:
Why force people to pass an immutable map?  Why not Map<String, String> ?  Internally, you can wrap it into something immutable, but I don't think it should matter from the constructor's perspective.